### DISCUSSION TITLE
FreeImage Dependency

### DISCUSSION THREAD

**Discussion by HerrNamenlos123** [openframeworks/openFrameworks#7840] [General]
@danoli3 I just took a look at FreeImage in the apothecary and found that a fork of yours is used. I also saw CMake support, which is awesome. I currently take many packages and build them using the new apothecary replacement i am prototyping with, and I want to give FreeImage a shot. If it turns out to compile without much hassle, it can later be easily made into a Conan package, so we have our own forked version of FreeImage in a custom Conan package. But that's for later.

Could you please answer a few questions so we are on the same page?
 - What has been changed in your FreeImage fork and why is it important? 
 - Why is your fork not used for VS and MSYS currently in the apothecary? 
 - Could we build and link your fork for ALL configurations including MSVC, MSYS and Emscripten?
 - And lastly, any important notes when I try to build it using CMake?

**Comment by HerrNamenlos123** [comment]
Wow I have to say I am impressed. A simple CMake call and FreeImage and dependencies *almost* built out of the box. Between 100000 lines of files being compiled there were only a handful of errors:
```
C:\Users\zachs\Projects\openFrameworks-packages\temp\freeimage\source\Source\LibRawLite\src\tables\wblists.cpp(217,31): warning C4273: 'Sony_SR2_wb_list1': inconsistent dll linkage [C:\Users\zachs\Projects\o 
penFrameworks-packages\temp\freeimage\build_debug\FreeImage.vcxproj]
C:\Users\zachs\Projects\openFrameworks-packages\temp\freeimage\source\Source\LibRawLite\internal/libraw_internal_funcs.h(52,31): message : see previous definition of 'protected: static libraw_static_table_t  
LibRaw::Sony_SR2_wb_list1' [C:\Users\zachs\Projects\openFrameworks-packages\temp\freeimage\build_debug\FreeImage.vcxproj]
C:\Users\zachs\Projects\openFrameworks-packages\temp\freeimage\source\Source\LibRawLite\src\tables\wblists.cpp(217,96): error C2491: 'LibRaw::Sony_SR2_wb_list1': definition of dllimport static data member no 
t allowed [C:\Users\zachs\Projects\openFrameworks-packages\temp\freeimage\build_debug\FreeImage.vcxproj]
```
Do you know what that could be?

**Comment by HerrNamenlos123** [comment]
Also I see you are actively implementing CMake support, that is great!
Now, I don't how much knowledge of CMake you have but if you want, 
I could give you a bit of constructive feedback on Do's and Dont's in CMake. If you want :)

**Reply by danoli3** [reply]
Yes please can you help us figure out wasm emscripten via cmake

**Reply by HerrNamenlos123** [reply]
How do you prefer the feedback? Should I just explain things so you can fix it yourself or should I fork, fix and create a pull request, and then afterwards explain why I changed it?

**Reply by HerrNamenlos123** [reply]
I created a pull request in your FreeImage repository. Feel free to merge it if it works. Remember that these changes are just to get CMake into a state it should be. Adding Emscripten should then be easier. If it still does not work, you need to tell me what exactly you want Emscripten to do and what doesn't work.
